### PR TITLE
MES-4427: WRTC Page & Vehicle Checks Modal for A Mod 2

### PR DESCRIPTION
**Pull request by szabikr**
## Description

- Creating initial screen for WRTC Page
- Creating initial screen for Vehicle Checks
- Adding Todos where needed

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Review by edwardwilson — APPROVED** [review]

File: src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        component.formGroup = formBuilder.group({
+          vehicleChecksSelectQuestions: null,
+        });
+        component.formControl = formBuilder.control({});
+        component.ngOnChanges();
+        expect(component.formControl.value).toEqual('Select questions');
+      });
+    });
+
+    describe('DOM', () => {
```
Comment (MarkPatt89) [inline src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts]: I know you didn't add it, it's elsewhere in the code. But can we get rid of these?

File: src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/vehicle-checks.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+@Component({
+  selector: 'vehicle-checks-cat-a-mod2',
+  templateUrl: 'vehicle-checks.html',
+})
+export class VehicleChecksCatAMod2Component implements OnChanges {
+
+  @Input() onCloseVehicleChecksModal: () => {};
+
+  @Input() vehicleChecksScore: VehicleChecksScore;
+  @Input() vehicleChecks: CatBEUniqueTypes.VehicleChecks;
```
Comment (MarkPatt89) [inline src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/vehicle-checks.ts:24]: Todo needed here.

File: src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/vehicle-checks.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import { VehicleChecksScore } from '../../../../../shared/models/vehicle-checks-score.model';
+
+// TODO - PREP-AMOD2: Use a mod2 types
+import { CatBEUniqueTypes } from '@dvsa/mes-test-schema/categories/BE';
+import { get } from 'lodash';
+import { QuestionResult } from '@dvsa/mes-test-schema/categories/common';
+
+@Component({
+  selector: 'vehicle-checks-cat-a-mod2',
+  templateUrl: 'vehicle-checks.html',
```
Comment (MarkPatt89) [inline src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/vehicle-checks.ts:15]: Is this correct? Is `vehicle-checks.html` shared throughout with the only difference being the selector?

File: src/pages/waiting-room-to-car/cat-a-mod2/waiting-room-to-car.cat-a-mod2.page.ts [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+  instructorAccompaniment$: Observable<boolean>;
+  supervisorAccompaniment$: Observable<boolean>;
+  otherAccompaniment$: Observable<boolean>;
+  interpreterAccompaniment$: Observable<boolean>;
+  eyesightTestComplete$: Observable<boolean>;
+  eyesightTestFailed$: Observable<boolean>;
+  gearboxAutomaticRadioChecked$: Observable<boolean>;
+  gearboxManualRadioChecked$: Observable<boolean>;
+  vehicleChecksScore$: Observable<VehicleChecksScore>;
+  vehicleChecks$: Observable<CatBEUniqueTypes.VehicleChecks>;
```
Comment (MarkPatt89) [inline src/pages/waiting-room-to-car/cat-a-mod2/waiting-room-to-car.cat-a-mod2.page.ts:107]: todo needed here

**Review by MarkPatt89 — COMMENTED** [review]
Just a few wee todos remaining, and a question around a `selector` and `templateURL` which may turn out to be nothing.

File: src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts (hunk 1 above)
Comment (szabikr) [reply src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts]: You want me to delete the `DOM` tests?

File: src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/vehicle-checks.ts (hunk 3 above)
Comment (szabikr) [reply src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/vehicle-checks.ts:15]: I think the selector shouldn't have the category in it, unless we want to include this alongside with another vehicle-checks to the same module. Would be nice to get to the bottom of this. But for consistency we can leave it as it is for now

Comment (MarkPatt89) [reply src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/vehicle-checks.ts:15]: Cheers for the clarification :)

File: src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts (hunk 1 above)
Comment (szabikr) [reply src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/__tests__/vehicle-checks.spec.ts]: deleted

File: src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/vehicle-checks.ts (hunk 2 above)
Comment (szabikr) [reply src/pages/waiting-room-to-car/cat-a-mod2/components/vehicle-checks/vehicle-checks.ts:24]: added

**Review by mgoldson — APPROVED** [review]

File: src/pages/waiting-room-to-car/cat-a-mod2/waiting-room-to-car.cat-a-mod2.page.ts (hunk 4 above)
Comment (szabikr) [reply src/pages/waiting-room-to-car/cat-a-mod2/waiting-room-to-car.cat-a-mod2.page.ts:107]: added

**Review by MarkPatt89 — APPROVED** [review]

**Pull request merged**